packages/components/field-errors/src/field-errors.tsx

```diff
@@ -1,18 +1,39 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { FormattedMessage } from 'react-intl';
 import { ErrorMessage } from '@commercetools-uikit/messages';
 import messages from './messages';
 
-const isObject = (obj) => typeof obj === 'object';
+const isObject = (obj: unknown): boolean => typeof obj === 'object';
 
-const FieldErrors = (props) => {
+type TErrorRenderer = (key: string, error?: boolean) => React.ReactNode;
+
+type TFieldErrorsProps = {
+  /**
+   * List of errors. Only entries with truthy values will count as active errors.
+   */
+  errors: Record<string, boolean>;
+  /**
+   * `true` when the error messages should be rendered. Usually you'd pass in a `touched` state of fields.
+   */
+  isVisible?: boolean;
+  /**
+   * Function which gets called with each error key (from the `errors` prop) and may render an error message or return `null` to hand the error handling off to `renderDefaultError`.
+   */
+  renderError?: TErrorRenderer;
+  /**
+   * Function which gets called with each error key (from the `errors` prop) for which `renderError` returned `null`.
+   * It may render an error message or return `null` to hand the error handling off to `FieldError`s built-in error handling.
+   */
+  renderDefaultError?: TErrorRenderer;
+};
+
+const FieldErrors = (props: TFieldErrorsProps) => {
   if (!props.isVisible) return null;
   if (!isObject(props.errors)) return null;
 
   return (
     <React.Fragment>
-      {Object.entries(props.errors)
+      {Object.entries<boolean>(props.errors)
```

Comment:
I guess TS is smart enough to infer `boolean` here, isn't it? Should work fine without explicit type parameter.

Reply:
oh.. I set it there because I wanted to test the `errors` object with different types.
but yes, the intellisense covers this :)